host-bmc/host_pdr_handler.cpp

```diff
@@ -1840,6 +1879,53 @@ void HostPDRHandler::setOperationStatus()
     }
 }
 
+std::optional<std::string_view>
+    HostPDRHandler::fetchPrettyName(const pldm_entity& entity)
+{
+    if (entityAuxiliaryNamesList.empty())
+    {
+        error("Entity auxiliary names list is empty");
+        return std::nullopt;
+    }
+
+    auto it = std::find_if(
+        entityAuxiliaryNamesList.begin(), entityAuxiliaryNamesList.end(),
+        [entity](
+            const std::shared_ptr<EntityAuxiliaryNames>& entityAuxiliaryNames) {
+        const auto& [key, entityNames] = *entityAuxiliaryNames;
+        return (entityAuxiliaryNames && key.type == entity.entity_type &&
+                key.instanceIdx == entity.entity_instance_num &&
+                key.containerId == entity.entity_container_id &&
+                entityNames.size());
```

Comment:
this will become simple if we use map for entityAuxiliaryNamesList.

Reply:
Same comment as above and not sure whether I will be able to upstream with a different way to parse the same PDR.